### PR TITLE
[lint, type] Linting, docstring and type improvements in `_` files

### PR DESCRIPTION
**Pull request by cidrblock**
depends-on: https://github.com/ansible/ansible-navigator/pull/671
- does what it says

### CODE REVIEW

File: src/ansible_navigator/__init__.py [hunk 1]
```diff
@@ -1,3 +1,2 @@
-""" init
-"""
+"""The initialization file for ansible-navigator."""
```
Comment (webknjaz) [inline src/ansible_navigator/__init__.py:1]: `__init__` represents a package (directory) docstring, not a single module (file). This is why such docstrings are expected to describe the package (as in a Python importable package, not a distribution package).

It'd be something like
```suggestion
"""Ansible-navigator TUI."""
```

You could optionally extend it with "doctest"-style examples or RST code block snippets. For example, look at this https://cheroot.cherrypy.dev/en/latest/pkg/cheroot.cli/ (although this specific example isn't for a package but for a single module, it should demonstrate the idea).

File: src/ansible_navigator/_yaml.py [hunk 2]
```diff
@@ -22,14 +22,20 @@
...
-    """the params for yaml dump"""
+    """The parameters for yaml dump."""
 
     default_flow_style: bool = False
     explicit_start: bool = True
 
 
 def human_dump(obj: Any, filename: str = None, fmode: str = "w") -> Union[str, None]:
-    """Consistant dumping of object across the application"""
+    """Consistant dumping of object across the application.
```
Comment (webknjaz) [inline src/ansible_navigator/_yaml.py:32]: 1) There's a typo in "Consistent"
2) Functions/methods represent actions and their docstrings should start with a verb. Does it do something or is it just a static concept you're describing?

```suggestion
    """Dump an object consistently across the application.
```

(you could optionally expand on what it actually means in a separate paragraph)

File: src/ansible_navigator/_yaml.py [hunk 3]
```diff
@@ -45,10 +51,16 @@ def human_dump(obj: Any, filename: str = None, fmode: str = "w") -> Union[str, N
 
 class HumanDumper(Dumper):
     # pylint: disable=too-many-ancestors
-    """for block scalar for multiline"""
+    """Dump a multiline in a human readable format."""
```
Comment (webknjaz) [inline src/ansible_navigator/_yaml.py:54]: This one is the exact opposite of my previous comment — it should describe a static thing/concept/object. By itself, a class instance does not do anything (unless it implements a `__call__()` method, that is; but even then, I'd put the action in the dunder-call docstring, not the class docstring), it does not dump a multiline; its methods do the dumping but merely initializing this object does not have side-effects as it currently implies.

*Hint:* for classes/constants/vars/attrs/properties/modules/packages, describe what they are, not what they "do". If there's a side effect of doing something during the initialization, something is seriously wrong with that design.

```suggestion
    """An implementation of YAML the dumping protocol

    It is designed with human readability in mind and its methods
    emit multiline string representation.
    """
```
(not sure how precise this is but I hope you got the idea)

File: src/ansible_navigator/_yaml.py [hunk 4]
```diff
@@ -62,8 +74,15 @@ def represent_scalar(self, tag, value, style=None):
         return super().represent_scalar(tag, value, style)
 
 
-# from http://stackoverflow.com/a/15423007/115478
-def _is_multiline_string(value):
+def _is_multiline_string(value: str):
+    """Determine if a string is multiline.
+
+    Note:
+    from http://stackoverflow.com/a/15423007/115478
```
Comment (webknjaz) [inline src/ansible_navigator/_yaml.py:81]: We can use an RST admonition for this:
```suggestion
    .. note::

       Inspired by http://stackoverflow.com/a/15423007/115478.
```

Comment (webknjaz) [reply src/ansible_navigator/_yaml.py:81]: Or maybe it's better to be
```suggestion
    .. tip::

       Inspired by http://stackoverflow.com/a/15423007/115478.
```
(or "hint"?)

File: src/ansible_navigator/_yaml.py [hunk 5]
```diff
@@ -62,8 +74,15 @@ def represent_scalar(self, tag, value, style=None):
...
+
+    :param value: The value to check
+    :return: a boolean indicating is the string is multiline
```
Comment (webknjaz) [inline src/ansible_navigator/_yaml.py:84]: Other docstrings have this capitalize. So it'd be consistent to do
```suggestion
    :return: A boolean indicating if the string is multiline
```
Also `s/is/if/`

File: src/ansible_navigator/__init__.py (hunk 1 above)
Comment (cidrblock) [reply src/ansible_navigator/__init__.py:1]: @webknjaz in this case the init file is for the ansible_navigator package (the entirety of the source)....  It probably should have been an underscore?

Comment (cidrblock) [reply src/ansible_navigator/__init__.py:1]: got it

File: src/ansible_navigator/_yaml.py (hunk 2 above)
Comment (cidrblock) [reply src/ansible_navigator/_yaml.py:32]: got it

File: src/ansible_navigator/_yaml.py (hunk 3 above)
Comment (cidrblock) [reply src/ansible_navigator/_yaml.py:54]: got it

File: src/ansible_navigator/_yaml.py (hunk 4 above)
Comment (cidrblock) [reply src/ansible_navigator/_yaml.py:81]: got it

File: src/ansible_navigator/_yaml.py (hunk 5 above)
Comment (cidrblock) [reply src/ansible_navigator/_yaml.py:84]: ty

**Comment by cidrblock** [comment]
recheck

File: src/ansible_navigator/_version.py [hunk 6]
```diff
@@ -1,7 +1,6 @@
-""" version
+"""Version information for ansible-navigator.
 
 Note:
```
Comment (webknjaz) [inline src/ansible_navigator/_version.py:3]: Looks like this could be an RST `.. note::` admonition.

File: src/ansible_navigator/_version.py [hunk 7]
```diff
@@ -1,7 +1,6 @@
...
-
```
Comment (webknjaz) [inline src/ansible_navigator/_version.py:4]: I wouldn't remove this line. Other vars below have a separate empty line before them. Plus if you use an admonition, the whole block would need to be indented.

File: src/ansible_navigator/_yaml.py [hunk 8]
```diff
@@ -45,10 +53,19 @@ def human_dump(obj: Any, filename: str = None, fmode: str = "w") -> Union[str, N
...
 class HumanDumper(Dumper):
     # pylint: disable=too-many-ancestors
-    """for block scalar for multiline"""
+    """An instance of a pyyaml Dumper.
+
+    This deviates from the base to dump a multiline string in a human readable format.
+    """
+
+    def represent_scalar(self, tag: str, value: str, style=Union[str, None]) -> yaml.ScalarNode:
+        """Use a block scalar for a nicer human representation of multiline strings.
```
Comment (webknjaz) [inline src/ansible_navigator/_yaml.py:62]: What does this do exactly? "use" does not specify a side-effect, is it a converter? Maybe it'd be "Convert one thing into another"?

File: src/ansible_navigator/_version.py (hunk 6 above)
Comment (cidrblock) [reply src/ansible_navigator/_version.py:3]: done

File: src/ansible_navigator/_version.py (hunk 7 above)
Comment (cidrblock) [reply src/ansible_navigator/_version.py:4]: the others have a blank after, not before :)

Comment (cidrblock) [reply src/ansible_navigator/_version.py:4]: put back

File: src/ansible_navigator/_yaml.py (hunk 8 above)
Comment (cidrblock) [reply src/ansible_navigator/_yaml.py:62]: updated

**Comment by cidrblock** [comment]
recheck

**Review by webknjaz — APPROVED** [review]

**Pull request merged**